draft-ietf-tls-esni.md

```diff
@@ -347,148 +408,92 @@ computed from Z as follows:
 
 ~~~~
    Zx = HKDF-Extract(0, Z)
-   key = HKDF-Expand-Label(Zx, KeyLabel, Hash(ESNIContents), key_length)
-   iv = HKDF-Expand-Label(Zx, IVLabel, Hash(ESNIContents), iv_length)
+   key = HKDF-Expand-Label(Zx, KeyLabel, ClientHelloOuter.Random, key_length)
+   iv = HKDF-Expand-Label(Zx, IVLabel, ClientHelloOuter.Random, iv_length)
 ~~~~
 
-where ESNIContents is as specified below and Hash is the hash function
-associated with the HKDF instantiation. The salt argument for HKDF-Extract is a
-string consisting of Hash.length bytes set to zeros. For a client's first
-ClientHello, KeyLabel = "esni key" and IVLabel = "esni iv", whereas for a
-client's second ClientHello, sent in response to a HelloRetryRequest,
-KeyLabel = "hrr esni key" and IVLabel = "hrr esni iv". (This label variance
-is done to prevent nonce re-use since the client's ESNI key share, and
-thus the value of Zx, does not change across ClientHello retries.)
-
-Note that ESNIContents will not be directly transmitted to the server in the
-ClientHello. The server will instead reconstruct the same object by obtaining
-its values from ClientEncryptedSNI and ClientHello.
+Where the Hash for HKDF is the hash function associated with the HKDF
+instantiation. The salt argument for HKDF-Extract is a string
+consisting of Hash.length bytes set to zeros. For a client's first
+ClientHello, KeyLabel = "esni key" and IVLabel = "esni iv", whereas
+for a client's second ClientHello, sent in response to a
+HelloRetryRequest, KeyLabel = "hrr esni key" and IVLabel = "hrr esni
+iv". (This label variance is done to prevent nonce re-use since the
+client's ESNI key share, and thus the value of Zx, does not change
+across ClientHello retries.)
 
 [[TODO: label swapping fixes a bug in the spec, though this may not be
 the best way to deal with HRR. See https://github.com/tlswg/draft-ietf-tls-esni/issues/121
 and https://github.com/tlswg/draft-ietf-tls-esni/pull/170 for more details.]]
 
-~~~
-   struct {
-       opaque record_digest<0..2^16-1>;
-       KeyShareEntry esni_key_share;
-       Random client_hello_random;
-   } ESNIContents;
-~~~
-
-record_digest
-: Same value as ClientEncryptedSNI.record_digest.
-
-esni_key_share
-: Same value as ClientEncryptedSNI.key_share.
+The client MAY replace any large, duplicated, extensions in ClientHelloInner
```

Comment:
I think the commas are off here. I think the best fix might be to leave out "large" and write "duplicated extensions".